Test task by Yaroslav Samoilenko | yaroslav.samoilenko.r@gmail.com

Xilston opened the pull request:
- added migration file to add city column
 - added seeders to add city to the user profile
 - added city displaying to the requiter profile
 - added ProfileCity component (it has title property for screen readers and the ability to open this city with Google Maps)
 
https://www.loom.com/share/459660a425c54ce7808d6ff6662cc516?sid=bcbbdbc9-9090-4e9e-878c-990312deeb81

ivadimko commented on api/db/migrations/.gitkeep:
why did you move it?

ivadimko commented on api/src/models/RecruiterProfile.ts:
```diff
@@ -67,4 +67,7 @@ export class RecruiterProfile extends ModelBase<RecruiterProfile> {
   deletedAt: Date
+  
+  @Column
+  city: string;
```
is it nullable?

Vlislave replied:
fixed

Xilston replied:
fixed

ivadimko commented on api/src/modules/recruiterProfile/recruiterProfile.fragments.serverless/tempCodeRunnerFile.ts:
what is the purpose of this file?

Xilston replied:
It seems like it was generated automatically, thanks for the note.

ivadimko commented on config/config.json:
why do you need this file?

ivadimko commented on frontend/src/components/Profile/ProfilePreview/ProfileMeta/ProfileCity/ProfileCity.module.scss:
```diff
@@ -0,0 +1,50 @@
+    margin-right: 0;
+  }
+
+  &--linked {
```
are dashes available while using in the component styles?
also, as we're using CSS modules, it's not obligatory to use BEM

Xilston replied:
> ivadimko

there are some tricks on how to use them (shown at screenshot, `linkedProfileClass` variable)
<img width="873" alt="image" src="https://github.com/mate-academy/huntd-public/assets/129086393/9e212693-1893-4425-a21c-69d24abcb3b5">

but I got your idea and I'll change it.

ivadimko commented on frontend/src/components/Profile/ProfilePreview/ProfileMeta/ProfileCity/ProfileCity.module.scss:
```diff
@@ -0,0 +1,50 @@
+    @extend %hover-underline;
+
+    .profileCity__icon {
+      animation: location_anim 3s linear infinite;
```
check animation variables in the project, let's make styles consistent

ivadimko commented on frontend/src/components/Profile/ProfilePreview/ProfileMeta/ProfileCity/ProfileCity.module.scss:
```diff
@@ -0,0 +1,50 @@
+  }
+}
+
+@keyframes location_anim {
```
what does `anim` mean here?
why not the full name?

ivadimko commented on frontend/src/components/Profile/ProfilePreview/ProfileMeta/ProfileCity/ProfileCity.tsx:
```diff
@@ -0,0 +1,82 @@
+      .then((response) => response.json())
+      .then((data) => {
+        const city = data.geonames[0];
+        // eslint-disable-next-line padding-line-between-statements, no-console
```
it could be better to fix what eslint says instead of disabling

ivadimko commented on frontend/src/components/Profile/ProfilePreview/ProfileMeta/ProfileCity/ProfileCity.tsx:
```diff
@@ -0,0 +1,82 @@
+}
+
+const geoService = {
+  url: 'https://secure.geonames.org/searchJSON?q=',
+  user: 'yasik2255',
+  params: '&maxRows=1&username=',
```
should it be hardcoded?

ivadimko commented on frontend/src/components/Profile/ProfilePreview/ProfileMeta/RecruiterProfileMeta.tsx:
```diff
@@ -17,7 +18,7 @@ export const RecruiterProfileMeta = React.memo<Props>((props) => {
           key={item.name}
         >
-          {item.text}
+          {item.name !== 'city' ? item.text : <ProfileCity title={item.text} />}
```
should we really rely on item name? what if we need an icon for another element? How can we make a code more re-usable?

ivadimko commented on package.json:
```diff
@@ -26,7 +26,9 @@
     "husky": "^4.3.8",
-    "lint-staged": "^10.2.11"
+    "lint-staged": "^10.2.11",
+    "mysql2": "^3.9.2",
+    "sequelize": "^6.37.1"
```
why do you need it here?

ivadimko commented on package.json:
```diff
@@ -42,13 +44,15 @@
     "stylelint-declaration-use-variable": "^1.7.2",
-    "stylelint-order": "^4.1.0"
+    "stylelint-order": "^4.1.0",
+    "typescript": "^5.4.3"
```
interesting how it worked before? why do you need all these packages?

ivadimko reviewed (commented):
let's focus on what is requried in test assignment and then add new features step by step:
- pay attention to linter issues
- please record video how you run migration and seeder
- make components easy to reuse, e.g. if we need icon on another component, it should be easy to add one

Xilston commented:
@ivadimko 
 - added `ProfileMetaItem` component to render meta item with the possibility to set icon (look at the screenshot)
 - added `@nullable` ability for the city column in schema
 - rewrited  `--lenked` class to `_linked`
 - moved `geoService` to a separate service folder &. created constants for its usage.
 - attached video with migrations & seeding

![telegram-cloud-photo-size-2-5199464778811168933-y](https://github.com/mate-academy/huntd-public/assets/129086393/db3cbe54-413e-4540-84ec-9026dc77db92)

https://www.loom.com/share/57386465ce9a4c5d870da4dffede2f64?sid=bda4085f-2503-4d6d-9ed0-f04810d61588

ivadimko commented on frontend/src/components/Profile/ProfilePreview/ProfileMeta/ProfileCity/ProfileCity.tsx:
```diff
@@ -0,0 +1,68 @@
+        if (city) {
+          setCityLink(geoService.getCityLink(city.name));
+        } else {
+          // eslint-disable-next-line no-console
```
linter errors still not fixed

ivadimko commented on package.json:
why do we have changes in this file?

Vlislave commented on api/.sequelizerc:
```diff
@@ -2,4 +2,5 @@ const path = require('path');
 
 module.exports = {
   'config': path.resolve('config', 'config.js'),
+  'migrations-path': path.resolve('db', 'migrations'),
```
in the process when I tried to figure out how to work with docker I tried some ways, but it is not need now, I'll fix it

Xilston replied:
I tried to figure out how to work with docker and tried some ways, it doesn't need now, I'll fix that

Vlislave commented on api/src/modules/recruiterProfile/recruiterProfile.fragments.serverless/tempCodeRunnerFile.ts:
```diff
@@ -0,0 +1 @@
+RecruiterProfile
```
looks like it's the autogenerated file, I'll delete it

Xilston replied:
looks like it is the autogenerated file, I'll delete it

Xilston commented:
@ivadimko I've done fixes.
 - restored base migrations directory
 - deleted autogenerated file
 - fixed linter issues
 - restored package.json to base statement
